tests/components/rest_command/test_init.py

```diff
@@ -43,6 +48,42 @@ def test_setup_component_test_service(self):
 
         assert self.hass.services.has_service(rc.DOMAIN, "test_get")
 
+    def test_reload(self):
+        """Verify we can reload rest_command integration."""
+
+        with assert_setup_component(1):
+            setup_component(self.hass, rc.DOMAIN, self.config)
+
+        assert self.hass.services.has_service(rc.DOMAIN, "test_get")
+        assert not self.hass.services.has_service(rc.DOMAIN, "new_test")
+
+        # Old config retained if rest_command is not in new config
+        with patch(
+            "homeassistant.config.load_yaml_config_file",
+            autospec=True,
+            return_value={},
+        ):
+            self.hass.services.call(rc.DOMAIN, SERVICE_RELOAD, blocking=True)
+
+        assert self.hass.services.has_service(rc.DOMAIN, "test_get")
+        assert not self.hass.services.has_service(rc.DOMAIN, "new_test")
```

Comment:
Oh, it looks like this was intended. Is this standard practice already?

Reply:
Not sure if standard practice, but copied from / inspired by the `Rest`, and `HomeKit` components.